app/code/Magento/Catalog/Model/CategoryList.php

```diff
@@ -48,25 +50,33 @@ class CategoryList implements CategoryListInterface
      */
     private $collectionProcessor;
 
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
     /**
      * @param CollectionFactory $categoryCollectionFactory
      * @param JoinProcessorInterface $extensionAttributesJoinProcessor
      * @param CategorySearchResultsInterfaceFactory $categorySearchResultsFactory
      * @param CategoryRepositoryInterface $categoryRepository
      * @param CollectionProcessorInterface|null $collectionProcessor
+     * @param StoreManagerInterface|null $storeManager
      */
     public function __construct(
         CollectionFactory $categoryCollectionFactory,
         JoinProcessorInterface $extensionAttributesJoinProcessor,
         CategorySearchResultsInterfaceFactory $categorySearchResultsFactory,
         CategoryRepositoryInterface $categoryRepository,
-        CollectionProcessorInterface $collectionProcessor = null
+        CollectionProcessorInterface $collectionProcessor = null,
+        StoreManagerInterface $storeManager = null
     ) {
         $this->categoryCollectionFactory = $categoryCollectionFactory;
         $this->extensionAttributesJoinProcessor = $extensionAttributesJoinProcessor;
         $this->categorySearchResultsFactory = $categorySearchResultsFactory;
         $this->categoryRepository = $categoryRepository;
         $this->collectionProcessor = $collectionProcessor ?: $this->getCollectionProcessor();
+        $this->storeManager = $storeManager ?: ObjectManager::getInstance()->get(StoreManagerInterface::class);
```

Comment:
It would be better to check it using null coalescing operator.

```suggestion
        $this->collectionProcessor = $collectionProcessor ?? $this->getCollectionProcessor();
        $this->storeManager = $storeManager ?? ObjectManager::getInstance()->get(StoreManagerInterface::class);
```

Reply:
Done. Thank you for review and help.